x-pack/plugins/security/server/saved_objects/secure_saved_objects_client_wrapper.ts

```diff
@@ -617,32 +614,43 @@ export class SecureSavedObjectsClientWrapper implements SavedObjectsClientContra
     return uniq(objects.map((o) => o.type));
   }
 
-  private async getNamespacesPrivilegeMap(namespaces: string[]) {
+  private async getNamespacesPrivilegeMap(
+    namespaces: string[],
+    previouslypreviouslyAuthorizedSpaceIds: string[]
+  ) {
+    const namespacesToCheck = namespaces.filter(
+      (namespace) => !previouslypreviouslyAuthorizedSpaceIds.includes(namespace)
+    );
+    const initialPrivilegeMap = previouslypreviouslyAuthorizedSpaceIds.reduce(
+      (acc, spaceId) => acc.set(spaceId, true),
+      new Map<string, boolean>()
+    );
+    if (namespacesToCheck.length === 0) {
+      return initialPrivilegeMap;
+    }
     const action = this.actions.login;
-    const checkPrivilegesResult = await this.checkPrivileges(action, namespaces);
+    const checkPrivilegesResult = await this.checkPrivileges(action, namespacesToCheck);
     // check if the user can log into each namespace
-    const map = checkPrivilegesResult.privileges.kibana.reduce(
-      (acc: Record<string, boolean>, { resource, authorized }) => {
-        // there should never be a case where more than one privilege is returned for a given space
-        // if there is, fail-safe (authorized + unauthorized = unauthorized)
-        if (resource && (!authorized || !acc.hasOwnProperty(resource))) {
-          acc[resource] = authorized;
-        }
-        return acc;
-      },
-      {}
-    );
+    const map = checkPrivilegesResult.privileges.kibana.reduce((acc, { resource, authorized }) => {
+      // there should never be a case where more than one privilege is returned for a given space
+      // if there is, fail-safe (authorized + unauthorized = unauthorized)
+      if (resource && (!authorized || !acc.hasOwnProperty(resource))) {
```

Comment:
Since you're using a Map, I expect you're looking for `has` instead of `hasOwnProperty`?

```suggestion
      if (resource && (!authorized || !acc.has(resource))) {
```

Reply:
Good catch. I also added a unit test to exercise this code path.